### ISSUE TITLE
NavTool returns pages from other sites at root level

### ISSUE THREAD

**Issue by dsolistorres** [dotCMS/core#16589]
Reported via Support: Ticket [98273](https://dotcms.zendesk.com/agent/tickets/98273)
Reproducible in dotCMS 5.0.2, dotCMS 5.1.5

NavTool returns pages marked as "Show on Menu" from other sites at the root level.

### Steps to reproduce the behavior:
1. Go to 'System -> Sites' and create 2 new sites: `site1.dotcms.com` and `site2.dotcms.com`
2. Add a new page to the first site at the root level: 'Site1 Page'. Enable the "Show on Menu" option for the page.
3. Add a new page to the second site at the root level: 'Site2 Page'. Enable the "Show on Menu" option for the page.
4. Add VTL code to the above pages to show the navigation at the root level for the site, using 'NavTool' viewtool method 'getNav("/")'. For example:
```
#set($list = $navtool.getNav("/"))
<div class="list-group sidebar-menu">  
	#foreach($n in $list)
		<a href="${n.href}" class="list-group-item #if($n.active)active#end">${n.title}</a>
    #end
</div>
```
5. Preview the page from the first site. The page from the second site is shown on the navigation menu. The page from the first site is also shown in the navigation menu shown in the second site.

### Expected behavior
When using method 'getNav("/")' from NavTool viewtool, only pages from the current site should be returned.

### Screenshots
Page from site 1 showing the page from site 2 in the navigation menu:
![image](https://user-images.githubusercontent.com/2894221/57895153-75ebc280-7807-11e9-89c7-92dbdebda8c7.png)

### Additional context
This has been reported by customer via Support: Ticket [98273](https://dotcms.zendesk.com/agent/tickets/98273)
The issue is causing problems for the navigation menu in a multi-site dotCMS instance.

**Comment by craigWagner99** [comment]
Still happening for 5.1.6

**Comment by wezell** [comment]
Needs Tests

**Comment by fmontes** [comment]
1. Need to make it into master and release-5.2.5 branch
2. Add tests
3. QA

**Comment by erickgonzalez** [comment]
PR: https://github.com/dotCMS/core/pull/17853

**Comment by bryanboza** [comment]
Fixed, tested on release-5.2.5 and now I'm unable to reproduce.
